autoload/cmake/terminal.vim

```diff
@@ -413,6 +413,17 @@ function! s:terminal.Close() abort
     endif
 endfunction
 
+" Toggle Vim-CMake console window.
+"
+function! s:terminal.Toggle() abort
+    call s:logger.LogDebug('Invoked: terminal.Toggle()')
+    if bufexists(l:self.console_buffer)
```

Comment:
I'm not sure this always results in the intended behavior. For instance, if the console buffer exists but the console window is not visible, I would expect `Toggle` to bring up the console window, as opposed to closing it (it is already closed anyway). Would you agree?

You should be able to observe what I mean if you run `:CMakeToggle` twice.

Reply:
Thanks for the feedback!

I've made an attempt at fixing it in 6010f6df16191be561811f2db936627ad5a5cf26. I _did_ try to use `wincmd hide` instead of `wincmd q`, but that spawned a bunch of odd behaviour. As said, I am very new to vimscript development and I might be loading the plugin incorrectly (I simply use `:so autoload/cmake.vim` and `:so autoload/cmake/terminal.vim`, which seems to get the job done most of the time).

Reply:
This version works indeed :)

However, we can make the logic of the function even a little more simple. To check whether the Vim-CMake window is open, you can just replicate what the `Open` function does:

https://github.com/cdelledonne/vim-cmake/blob/8a1699f4c48796bfb9f55813036998c94bb0d533/autoload/cmake/terminal.vim#L361-L362

This results in a single if check, without the need for an additional nested check.

One more thing: we should pass `v:false` when calling `Open`, otherwise the existing Vim-CMake buffer gets deleted, which is not what we want when just toggling the window. Does it make sense?